central/globaldb/v2backuprestore/formats/postgresv1/postgres.go

```diff
@@ -2,23 +2,66 @@ package postgresv1
 
 import (
 	"io"
+	"math"
+	"strconv"
 
 	"github.com/pkg/errors"
 	"github.com/stackrox/rox/central/globaldb/v2backuprestore/common"
 	"github.com/stackrox/rox/central/globaldb/v2backuprestore/restore"
 	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/migrations"
+	"github.com/stackrox/rox/pkg/postgres/pgadmin"
+	"github.com/stackrox/rox/pkg/postgres/pgconfig"
 )
 
 var (
 	log = logging.LoggerForModule()
 )
 
 func restorePostgresDB(ctx common.RestoreFileContext, fileReader io.Reader, size int64) error {
-	log.Debugf("restorePostgresDB - dump size = %d", size)
+	log.Debug("restorePostgresDB")
 	err := restore.LoadRestoreStream(fileReader)
 	if err != nil {
 		return errors.Wrap(err, "unable to restore postgres")
 	}
 
 	return nil
 }
+
+func checkPostgresSize(ctx common.RestoreFileContext, fileReader io.Reader, size int64) error {
+	log.Debugf("checkPostgresSize -- check file size = %d", size)
+
+	bytes := make([]byte, size)
+
+	bytesRead, err := fileReader.Read(bytes)
+	if int64(bytesRead) < size || (err != nil && err != io.EOF) {
+		log.Warnf("Could not determine free disk space for Postgres: %v. Assuming free space is sufficient.", err)
+		return nil
+	}
+
+	restoreBytes, err := strconv.ParseInt(string(bytes[:]), 10, 64)
+	if err != nil {
+		log.Warnf("Could not determine free disk space for Postgres: %v. Assuming free space is sufficient.", err)
+		return nil
+	}
+
+	requiredBytes := int64(math.Ceil(float64(restoreBytes) * (1.0 + migrations.CapacityMarginFraction)))
+
+	_, dbConfig, err := pgconfig.GetPostgresConfig()
+	if err != nil {
+		return errors.Wrap(err, "Could not parse postgres config")
+	}
+
+	availableBytes, err := pgadmin.GetRemainingCapacity(dbConfig)
+	if err != nil {
+		log.Warnf("Could not determine free disk space for Postgres: %v. Assuming free space is sufficient for %d bytes.", err, requiredBytes)
+		return nil
+	}
+
+	hasSpace := float64(availableBytes) > float64(requiredBytes)
+	if !hasSpace {
```

Comment:
Are we going to have a more precise number or precise estimate on capacity soon? If we could not get a better number, we'd better not blocking restore process.

Reply:
@c-du I guess some of that depends on how you define soon.  I'd say that this is our best guess for 3.72.  For 3.73, we should spend some time on [ROX-12059](https://issues.redhat.com/browse/ROX-12059) and decide how we are going to arrive at that.  If you like, I could put a todo here and log a warning instead of returning the error.